crates/block-producer/src/block_builder/mod.rs

```diff
@@ -143,3 +148,31 @@ where
         Ok(())
     }
 }
+
+struct BlockProducer<BB> {
+    pub mempool: Arc<Mutex<Mempool>>,
+    pub block_interval: tokio::time::Duration,
+    pub block_builder: BB,
+}
```

Comment:
Is the plan to eventually get rid of the `BlockBuilder` trait and merge the functionality into this struct?

Also, should we add `simulated_proof_time` here too?

Reply:
Correct. And yes we should have some simulated time as well; thanks.